src/transformers/modeling_tf_utils.py

```diff
@@ -2047,29 +2048,6 @@ def register_for_auto_class(cls, auto_class="TFAutoModel"):
         cls._auto_class = auto_class
 
 
-def shape_list(tensor: Union[tf.Tensor, np.ndarray]) -> List[int]:
```

Comment:
This was defined in three different places previously. Instead of copy-pasting this function a for a fourth time, I moved it to `tf_utils.py` along with `set_tensor_by_indices_to_value`

Reply:
This is the reason why so many files are changed in this PR. It was quite entangled in the changes of this PR so I hope it's ok to have this refactor be included here.

Reply:
Ok for me, thanks for taking care of it

Reply:
I trust you to have moved everything needed along with it. Could you make a good first issue with other functions we could move to that new module? Similarly with `pytorch_utils`, we would only keep the classes for TFPreTrainedModel and models in `modeling_tf_utils` and have all util functions in `tf_utils`.